### DISCUSSION TITLE
Can't set initial position of camera with PhantomCamera2D?

### DISCUSSION THREAD

**Discussion by tsmuse** [ramokz/phantom-camera#429] [Question / Help Wanted]
When I set the follow target the position of the PhantomCamera2d and the associated Camera2D nodes is auto set for me by the script and cannot be changes. When I have "tween on load" enabled, it appears to start that camera move from global 0,0 no matter what I do. Am I missing something? Is there a way to set the starting point of that movement? I want the camera to move in, but not from the top left corner of the scene. 

Thanks :)

**Comment by ramokz** [comment]
No, you're not missing something.
It happens because when a node is instantiated into a scene, it spawns at the coordinate `0,0` before teleporting to its designated place. 

This is only happening in a tiny window, but the `PCamHost` is grabbing that `0,0` of the `PCam` coordinate, nonetheless, which it shouldn't.

It's just a minor tweak to `PCamHost` to make it wait until the node is fully instantiated into the scene with its correct coordinate before grabbing it. Can try it out by adding the following line below on line `1070` inside `phantom_camera_host.gd`:

```gdscript
if not pcam.tween_on_load:
  pcam.set_tween_skip(self, true)
if not pcam.is_node_ready(): await pcam.ready # Add this line
_find_pcam_with_highest_priority()

```

Note, will be adding this change to the next minor release.

In practice, it means that the active `PCam` of a scene, as set from the editor, will not trigger `Tween on Load` as nothing will, or can for that matter, tween to it as it's the first node in the tween flow.
